UI/auth-youtube.cpp

```diff
@@ -350,6 +350,36 @@ std::shared_ptr<Auth> YoutubeAuth::Login(QWidget *owner,
 }
 
 #ifdef BROWSER_AVAILABLE
+void YoutubeChatDock::timerEvent(QTimerEvent *event)
+{
+	if (event->timerId() != awaitLoginTimer)
+		return;
+
+	QPointer<YoutubeChatDock> this_ = this;
```

Comment:
What is this intended to accomplish? Usually capturing `this` by itself in the lambda should be enough. If there is the concern that the pointer might become invalid in between triggers of the timer, use the `*this` capture as suggested by ISO C++ and available since C++17 (which is OBS' lower version boundary).

Note that `*this` will create a copy of all class data members, but that is the only way to ensure that `this` can refer to valid data when the lambda is executed.

Reply:
Was also following existing precedent here: 
* https://github.com/obsproject/obs-studio/blob/ce6f9a4742b40b04e68ef759e0698bd5eac4360e/UI/auth-twitch.cpp#L442

Please let me know if you still prefer capturing `this` in the lambda directly.

Reply:
I discussed this internally and it's fine to re-use the pattern used in the other class, as it's safer versus introducing a different way to handle data capture for the lambda here.

Reply:
Okay, thanks. Resolving.